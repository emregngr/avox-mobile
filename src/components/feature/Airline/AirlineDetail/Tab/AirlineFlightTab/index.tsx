import React, { useMemo } from 'react'
import { View } from 'react-native'

import { Alliance } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/Alliance'
import { DestinationList } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/DestinationList'
import { RoutesList } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/RoutesList'
import { StatsGrid } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/StatsGrid'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { Airline } from '@/types/feature/airline'

interface AirlineFlightTabProps {
  airlineData: Airline
}

export const AirlineFlightTab = ({ airlineData }: AirlineFlightTabProps) => {
  const { selectedTheme } = useThemeStore()
  const { selectedLocale } = useLocaleStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { network, operations } = airlineData ?? {}
  const { alliance } = operations ?? {}
  const {
    destinationCount,
    destinationCountries,
    destinations,
    domesticConnections,
    internationalConnections,
    routes,
  } = network ?? {}

  const memoizedStats = useMemo(
    () => ({
      destinationCount,
      destinationCountries,
      domesticConnections,
      internationalConnections,
    }),
    [destinationCount, destinationCountries, domesticConnections, internationalConnections],
  )

  const iconColor = useMemo(() => colors.onPrimary100, [colors.onPrimary100])

  const localeStrings = useMemo(
    () => ({
      activeRoutes: getLocale('activeRoutes'),
      alliance: getLocale('alliance'),
      country: getLocale('country'),
      domesticDestinations: getLocale('domesticDestinations'),
      internationalDestination: getLocale('internationalDestination'),
      mainDestinations: getLocale('mainDestinations'),
      totalDestination: getLocale('totalDestination'),
    }),
    [selectedLocale],
  )

  return (
    <View className="px-4 gap-y-4">
      <StatsGrid
        localeStrings={{
          country: localeStrings.country,
          domesticDestinations: localeStrings.domesticDestinations,
          internationalDestination: localeStrings.internationalDestination,
          totalDestination: localeStrings.totalDestination,
        }}
        iconColor={iconColor}
        stats={memoizedStats}
      />

      <DestinationList
        destinations={destinations}
        iconColor={iconColor}
        title={localeStrings.mainDestinations}
      />

      <RoutesList iconColor={iconColor} routes={routes} title={localeStrings.activeRoutes} />

      {alliance !== 'none' && (
        <Alliance alliance={alliance} iconColor={iconColor} title={localeStrings.alliance} />
      )}
    </View>
  )
}
