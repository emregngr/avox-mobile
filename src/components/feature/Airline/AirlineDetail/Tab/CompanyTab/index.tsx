import React from 'react'
import { View } from 'react-native'

import { Company } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Company'
import { Contact } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Contact'
import { Hub } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Hub'
import { Map } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Map'
import { SocialMedia } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/SocialMedia'
import type { Airline } from '@/types/feature/airline'

interface CompanyTabProps {
  airlineData: Airline
}

export const CompanyTab = ({ airlineData }: CompanyTabProps) => {
  const { companyInfo, operations } = airlineData

  return (
    <View className="px-4 gap-y-4">
      <Company airlineData={airlineData} />

      <Contact companyInfo={companyInfo} operations={operations} />

      <SocialMedia companyInfo={companyInfo} />

      <Hub operations={operations} />

      <Map airlineData={airlineData} />
    </View>
  )
}
