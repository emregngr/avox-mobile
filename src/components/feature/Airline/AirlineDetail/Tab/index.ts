export * from './AirlineFlightTab'
export * from './CompanyTab'
export * from './FleetTab'
export * from './SafetyEnvTab'
