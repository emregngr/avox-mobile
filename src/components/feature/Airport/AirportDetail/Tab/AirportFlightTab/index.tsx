import { useMemo } from 'react'
import { View } from 'react-native'

import { AirlinesList } from '@/components/feature/Airport/AirportDetail/Tab/AirportFlightTab/Sections/AirlinesList'
import { RoutesList } from '@/components/feature/Airport/AirportDetail/Tab/AirportFlightTab/Sections/RoutesList'
import { StatsGrid } from '@/components/feature/Airport/AirportDetail/Tab/AirportFlightTab/Sections/StatsGrid'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { AirportType } from '@/types/feature/airport'

interface AirportFlightTabProps {
  airportData: AirportType
}

export const AirportFlightTab = ({ airportData }: AirportFlightTabProps) => {
  const { selectedLocale } = useLocaleStore()
  const { selectedTheme } = useThemeStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { flightOperations } = airportData ?? {}
  const {
    airlines,
    destinationCount,
    destinationCountries,
    domesticConnections,
    internationalConnections,
    routes,
  } = flightOperations ?? {}

  const memoizedStats = useMemo(
    () => ({
      destinationCount,
      destinationCountries,
      domesticConnections,
      internationalConnections,
    }),
    [destinationCount, destinationCountries, domesticConnections, internationalConnections],
  )

  const localeStrings = useMemo(
    () => ({
      airlines: getLocale('airlines'),
      country: getLocale('country'),
      domesticDestinations: getLocale('domesticDestinations'),
      internationalDestination: getLocale('internationalDestination'),
      popularDestinations: getLocale('popularDestinations'),
      totalDestination: getLocale('totalDestination'),
    }),
    [selectedLocale],
  )

  return (
    <View className="px-4 gap-y-4">
      <StatsGrid
        localeStrings={{
          country: localeStrings.country,
          domesticDestinations: localeStrings.domesticDestinations,
          internationalDestination: localeStrings.internationalDestination,
          totalDestination: localeStrings.totalDestination,
        }}
        iconColor={colors?.onPrimary100}
        stats={memoizedStats}
      />

      <AirlinesList
        airlines={airlines}
        iconColor={colors?.onPrimary100}
        title={localeStrings.airlines}
      />

      <RoutesList
        iconColor={colors?.onPrimary100}
        routes={routes}
        title={localeStrings.popularDestinations}
      />
    </View>
  )
}
