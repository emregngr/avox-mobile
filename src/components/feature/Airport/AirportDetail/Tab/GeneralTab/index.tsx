import React from 'react'
import { View } from 'react-native'

import { Contact } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Contact'
import { General } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/General'
import { Location } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Location'
import { Map } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Map'
import { Services } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Services'
import { SocialMedia } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/SocialMedia'
import type { Airport } from '@/types/feature/airport'

interface GeneralTabProps {
  airportData: Airport
}

export const GeneralTab = ({ airportData }: GeneralTabProps) => {
  const { airportInfo, facilities, operations } = airportData ?? {}

  return (
    <View className="px-4 gap-y-4">
      <General airportInfo={airportInfo} operations={operations} />

      <Services facilities={facilities} />

      <Contact airportInfo={airportInfo} />

      <SocialMedia airportInfo={airportInfo} />

      <Location operations={operations} />

      <Map airportData={airportData} />
    </View>
  )
}
