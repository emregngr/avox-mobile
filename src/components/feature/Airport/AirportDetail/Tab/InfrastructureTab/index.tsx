import React from 'react'
import { View } from 'react-native'

import { Cargo } from '@/components/feature/Airport/AirportDetail/Tab/InfrastructureTab/Sections/Cargo'
import { Certifications } from '@/components/feature/Airport/AirportDetail/Tab/InfrastructureTab/Sections/Certifications'
import { Facilities } from '@/components/feature/Airport/AirportDetail/Tab/InfrastructureTab/Sections/Facilities'
import { Runway } from '@/components/feature/Airport/AirportDetail/Tab/InfrastructureTab/Sections/Runway'
import { Terminal } from '@/components/feature/Airport/AirportDetail/Tab/InfrastructureTab/Sections/Terminal'
import type { Airport } from '@/types/feature/airport'

interface InfrastructureTabProps {
  airportData: Airport
}

export const InfrastructureTab = ({ airportData }: InfrastructureTabProps) => {
  const { cargo, facilities, infrastructure, safety } = airportData || {}

  return (
    <View className="px-4 gap-y-4">
      <Terminal infrastructure={infrastructure} />

      <Runway infrastructure={infrastructure} />

      <Facilities facilities={facilities} infrastructure={infrastructure} />

      <Cargo cargo={cargo} />

      <Certifications safety={safety} />
    </View>
  )
}
