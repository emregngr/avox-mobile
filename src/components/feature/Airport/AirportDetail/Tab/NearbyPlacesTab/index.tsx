import React, { useCallback, useMemo } from 'react'
import { View } from 'react-native'

import { AttractionCard } from '@/components/feature/Airport/AirportDetail/Tab/NearbyPlacesTab/Cards/AttractionCard'
import { useMapNavigation } from '@/hooks/maps/useMapNavigation'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import type { Airport } from '@/types/feature/airport'

interface NearbyPlacesTabProps {
  airportData: Airport
}
interface AttractionCard {
  attractionId: number
  attractionLatitude: number
  attractionLongitude: number
  attractionName: string
  description: string
  formattedDistance: string
  handleDirectionPress: () => void
}

export const NearbyPlacesTab = ({ airportData }: NearbyPlacesTabProps) => {
  const { selectedLocale } = useLocaleStore()

  const { nearbyAttractions } = airportData || {}

  const localeStrings = useMemo(
    () => ({
      getDirection: getLocale('getDirection'),
      km: getLocale('km'),
    }),
    [selectedLocale],
  )

  const mapLocaleStrings = useMemo(
    () => ({
      appleMaps: getLocale('appleMaps'),
      cancel: getLocale('cancel'),
      googleMaps: getLocale('googleMaps'),
      selectMapApp: getLocale('selectMapApp'),
    }),
    [selectedLocale],
  )

  const { openMapNavigation } = useMapNavigation(mapLocaleStrings)

  const attractionItems: AttractionCard[] = useMemo(
    () =>
      nearbyAttractions?.map(attraction => {
        const {
          attractionCoordinates: { attractionLatitude, attractionLongitude },
          attractionId,
          attractionName,
          description,
          distanceKm,
        } = attraction

        const handleDirectionPress = () => {
          openMapNavigation(attractionLatitude, attractionLongitude, attractionName)
        }

        const formattedDistance = `${distanceKm} ${localeStrings.km}`

        return {
          attractionId,
          attractionLatitude,
          attractionLongitude,
          attractionName,
          description,
          formattedDistance,
          handleDirectionPress,
        }
      }) || [],
    [nearbyAttractions, openMapNavigation, localeStrings.km],
  )

  const renderAttraction = useCallback(
    (item: AttractionCard) => (
      <AttractionCard
        attractionId={item.attractionId}
        attractionName={item.attractionName}
        description={item.description}
        formattedDistance={item.formattedDistance}
        getDirectionText={localeStrings.getDirection}
        handleDirectionPress={item.handleDirectionPress}
        key={item.attractionId}
      />
    ),
    [localeStrings.getDirection],
  )

  return <View className="px-4 gap-y-4">{attractionItems?.map(renderAttraction)}</View>
}
