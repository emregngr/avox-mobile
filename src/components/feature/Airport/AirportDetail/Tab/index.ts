export * from './AirportFlightTab'
export * from './GeneralTab'
export * from './InfrastructureTab'
export * from './NearbyPlacesTab'
