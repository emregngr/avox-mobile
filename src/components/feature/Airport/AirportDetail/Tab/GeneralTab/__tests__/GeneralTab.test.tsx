import { render } from '@testing-library/react-native'
import React from 'react'

import { GeneralTab } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab'

const mockedGeneral = jest.fn()
const mockedServices = jest.fn()
const mockedContact = jest.fn()
const mockedSocialMedia = jest.fn()
const mockedLocation = jest.fn()
const mockedMap = jest.fn()

jest.mock('@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/General', () => {
  const { Text } = require('react-native')
  return {
    General: (props: any) => {
      mockedGeneral(props)
      return <Text testID="general-section">General</Text>
    },
  }
})

jest.mock('@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Services', () => {
  const { Text } = require('react-native')
  return {
    Services: (props: any) => {
      mockedServices(props)
      return <Text testID="services-section">Services</Text>
    },
  }
})

jest.mock('@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Contact', () => {
  const { Text } = require('react-native')
  return {
    Contact: (props: any) => {
      mockedContact(props)
      return <Text testID="contact-section">Contact</Text>
    },
  }
})

jest.mock('@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/SocialMedia', () => {
  const { Text } = require('react-native')
  return {
    SocialMedia: (props: any) => {
      mockedSocialMedia(props)
      return <Text testID="social-media-section">SocialMedia</Text>
    },
  }
})

jest.mock('@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Location', () => {
  const { Text } = require('react-native')
  return {
    Location: (props: any) => {
      mockedLocation(props)
      return <Text testID="location-section">Location</Text>
    },
  }
})

jest.mock('@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Map', () => {
  const { Text } = require('react-native')
  return {
    Map: (props: any) => {
      mockedMap(props)
      return <Text testID="map-section">Map</Text>
    },
  }
})

const mockedAirportData: any = {
  airportInfo: {
    id: 'IST',
    name: 'Istanbul Airport',
    email: 'info@istairport.com',
  },
  facilities: {
    lounges: 12,
    terminals: 1,
    shops: 200,
  },
  operations: {
    airportType: 'mega_airport',
    is24Hour: true,
  },
  statistics: {
    passengerCount: 70000000,
  },
}

describe('GeneralTab Component', () => {
  it('should render all section components', () => {
    const { getByTestId } = render(<GeneralTab airportData={mockedAirportData} />)

    expect(getByTestId('general-section')).toBeTruthy()
    expect(getByTestId('services-section')).toBeTruthy()
    expect(getByTestId('contact-section')).toBeTruthy()
    expect(getByTestId('social-media-section')).toBeTruthy()
    expect(getByTestId('location-section')).toBeTruthy()
    expect(getByTestId('map-section')).toBeTruthy()
  })

  it('should pass correct props to General component', () => {
    render(<GeneralTab airportData={mockedAirportData} />)

    expect(mockedGeneral).toHaveBeenCalledWith({
      airportInfo: mockedAirportData.airportInfo,
      operations: mockedAirportData.operations,
    })
  })

  it('should pass correct props to Services component', () => {
    render(<GeneralTab airportData={mockedAirportData} />)

    expect(mockedServices).toHaveBeenCalledWith({
      facilities: mockedAirportData.facilities,
    })
  })

  it('should pass correct props to Contact component', () => {
    render(<GeneralTab airportData={mockedAirportData} />)

    expect(mockedContact).toHaveBeenCalledWith({
      airportInfo: mockedAirportData.airportInfo,
    })
  })

  it('should pass correct props to SocialMedia component', () => {
    render(<GeneralTab airportData={mockedAirportData} />)

    expect(mockedSocialMedia).toHaveBeenCalledWith({
      airportInfo: mockedAirportData.airportInfo,
    })
  })

  it('should pass correct props to Location component', () => {
    render(<GeneralTab airportData={mockedAirportData} />)

    expect(mockedLocation).toHaveBeenCalledWith({
      operations: mockedAirportData.operations,
    })
  })

  it('should pass correct props to Map component', () => {
    render(<GeneralTab airportData={mockedAirportData} />)

    expect(mockedMap).toHaveBeenCalledWith({
      airportData: mockedAirportData,
    })
  })

  it('should handle undefined airportData gracefully', () => {
    render(<GeneralTab airportData={undefined as any} />)

    expect(mockedGeneral).toHaveBeenCalledWith({
      airportInfo: undefined,
      operations: undefined,
    })
    expect(mockedServices).toHaveBeenCalledWith({ facilities: undefined })
    expect(mockedMap).toHaveBeenCalledWith({ airportData: undefined })
  })

  it('should handle partial airportData (missing facilities)', () => {
    const partialData: any = {
      airportInfo: mockedAirportData.airportInfo,
      operations: mockedAirportData.operations,
    }

    render(<GeneralTab airportData={partialData} />)

    expect(mockedGeneral).toHaveBeenCalledWith({
      airportInfo: partialData.airportInfo,
      operations: partialData.operations,
    })
    expect(mockedServices).toHaveBeenCalledWith({ facilities: undefined })
  })

  it('should handle partial airportData (missing airportInfo)', () => {
    const partialData: any = {
      facilities: mockedAirportData.facilities,
      operations: mockedAirportData.operations,
    }

    render(<GeneralTab airportData={partialData} />)

    expect(mockedGeneral).toHaveBeenCalledWith({
      airportInfo: undefined,
      operations: partialData.operations,
    })
    expect(mockedContact).toHaveBeenCalledWith({ airportInfo: undefined })
    expect(mockedSocialMedia).toHaveBeenCalledWith({ airportInfo: undefined })
  })

  it('should call all section components exactly once', () => {
    render(<GeneralTab airportData={mockedAirportData} />)

    expect(mockedGeneral).toHaveBeenCalledTimes(1)
    expect(mockedServices).toHaveBeenCalledTimes(1)
    expect(mockedContact).toHaveBeenCalledTimes(1)
    expect(mockedSocialMedia).toHaveBeenCalledTimes(1)
    expect(mockedLocation).toHaveBeenCalledTimes(1)
    expect(mockedMap).toHaveBeenCalledTimes(1)
  })
})

describe('GeneralTab Component Snapshot', () => {
  it('should render the GeneralTab Component successfully', () => {
    const { toJSON } = render(<GeneralTab airportData={mockedAirportData} />)

    expect(toJSON()).toMatchSnapshot()
  })
})
