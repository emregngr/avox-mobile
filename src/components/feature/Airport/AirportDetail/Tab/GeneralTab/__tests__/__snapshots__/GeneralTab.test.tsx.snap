// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`GeneralTab Component Snapshot should render the GeneralTab Component successfully 1`] = `
<View
  className="px-4 gap-y-4"
>
  <Text
    testID="general-section"
  >
    General
  </Text>
  <Text
    testID="services-section"
  >
    Services
  </Text>
  <Text
    testID="contact-section"
  >
    Contact
  </Text>
  <Text
    testID="social-media-section"
  >
    SocialMedia
  </Text>
  <Text
    testID="location-section"
  >
    Location
  </Text>
  <Text
    testID="map-section"
  >
    Map
  </Text>
</View>
`;
